### ISSUE TITLE
Add a field for user data/ metadata

### ISSUE THREAD

**Issue by mostaphaRoudsari** [pollination/simulation-service#8]
This came up during a call today and will be very useful for filtering the simulations/ workflows/ objects based on user input data. It won't be used as a resource for any of Pollination tasks but it can be accessed by users or ideally should be possible to be used as an input query for `GET` calls.

Here is an example. The user sets up a parametric study from pollination. The parameters will be exposed in model and not directly in simulation. Now they can pass the important parameters from their model as metadata to simulation which means they can use it in the future to access those simulations.

Another case can be breaking down the simulations for multi-phase daylight simulation where they can tag each simulation using user_data filed.

**Comment by AntoineDao** [comment]
This is definitely going to be needed. Probably post MVP to be honest. However I reckon we could implement this by adding a `parameters` key in the simulation object that would then run a parameter extraction job that would be bound to each individual model compute in the results table. Here's an example:

```json
{
  "type": "DaylightFactor",
  "models": [ a, b ,c],
  "grids": [1, 2, 3],
  "parameters": [
    "glazing_ratio",
    "orientation" // not that it matters for daylight factor but you get my point
  ]
}
```

**Comment by mostaphaRoudsari** [comment]
Both `parameters` and `metrics` are nice but are different from what `user_data` are supposed to do. `user_data` will be custom data by user which will be ignored by Pollination during runs by default. What you are suggesting is a brute-force workflow similar to what Colibri does. Both are great and useful but different from each other.

**Comment by AntoineDao** [comment]
Yep makes sense. I guess we can create a `tags` list in the payload that takes key value pairs?
